### ISSUE TITLE
Pyramid view detection doesn't take grid mode into account

### ISSUE THREAD

**Issue by jni** [napari/napari#1388]
## 🐛 Bug

You're gonna love this one, @sofroniewn! 😂 Co-discovered by @DragaDoncila. 

## To Reproduce

```python
import numpy as np
import napari
pyramid0 = [np.random.random((4096 // 2**i, 4096 // 2**i)) for i in range(5)]
pyramid1 = [np.random.random((4096 // 2**i, 4096 // 2**i)) for i in range(5)]
v = napari.Viewer()
v.add_image(pyramid0, cmap='green')
v.add_image(pyramid0, colormap='green')
v.add_image(pyramid1, colormap='magenta', blending='additive')
```

Then enter grid mode and pan around.

## Actual behavior

![napari-pyramid-bug](https://user-images.githubusercontent.com/492549/85840775-639cd000-b7e0-11ea-88ff-7594858dd73e.gif)

## Expected behavior

Pyramid data should work fine in grid mode.

**Comment by sofroniewn** [comment]
Sigh, indeed, this doesn't surprise me. Will add to my To Do's :-) 

I want to pull out the grid stuff from the layer itself, but we don't have a nice place to put it now. In the sort term the fix is probably just needing to make sure some coordinate info passes through the inverse of the grid transform, but I'll need to check to be sure

**Comment by sofroniewn** [comment]
This has been closed by #1673
